src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/PhraseResultParser.kt

```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.hmpps.offendersearch.services
+
+import org.elasticsearch.search.SearchHit
+import uk.gov.justice.hmpps.offendersearch.dto.OffenderDetail
+
+
+internal fun extractOffenderDetailList(
+    hits: Array<SearchHit>,
+    phrase: String,
+    offenderParser: (json: String) -> OffenderDetail
+): List<OffenderDetail> {
+  return hits.map { offenderParser(it.sourceAsString).mergeHighlights(it.highlightFields, phrase) }
```

Comment:
Not sure it'll make it any cleaner but you could inline this.  Something like:
`List<OffenderDetail> : hits.map { offenderParser(it.sourceAsString).mergeHighlights(it.highlightFields, phrase) }`

Reply:
agreed, next PR